include/pybind11/pytypes.h

```diff
@@ -380,20 +381,39 @@ class PYBIND11_EXPORT_EXCEPTION error_already_set : public std::runtime_error {
 public:
     /// Constructs a new exception from the current Python error indicator, if any.  The current
     /// Python error indicator will be cleared.
-    error_already_set() : std::runtime_error(detail::error_string()) {
+    error_already_set() : std::runtime_error("") {
         PyErr_Fetch(&m_type.ptr(), &m_value.ptr(), &m_trace.ptr());
+        if (m_type) {
+            PyErr_NormalizeException(&m_type.ptr(), &m_value.ptr(), &m_trace.ptr());
+        }
     }
 
     error_already_set(const error_already_set &) = default;
     error_already_set(error_already_set &&) = default;
 
     inline ~error_already_set() override;
 
+    const char *what() const noexcept override {
+        if (m_lazy_what.empty()) {
+            try {
```

Comment:
I really wonder if this try catch is still necessary.

Reply:
I suggested try-catch block earlier because `detail::error_string(...)` call is not  `noexcept`. It's still the case.